src/Workspaces/Core/Portable/Utilities/EditDistance.cs

```diff
@@ -47,572 +68,581 @@ private static char[] ConvertToLowercaseArray(string text)
+            // was '2', then it would be impossible for the path from 'x' to provide a good
+            // enough edit distance *ever*.   Similarly:
+            //
+            //           s u n d a y
+            //      ----------------
+            //      |∞ ∞ ∞ ∞ ∞ ∞ ∞ ∞
+            //      |∞ 0 1 2 3 4 5 6
+            //    s |∞ 1
+            //    a |∞ 2
+            //    t |∞ 3 `
+            //    u |∞ 4   `
+            //    r |∞ 5     `
+            //    d |∞ 6       `
+            //    a |∞ 7         `
+            //    y |∞ 8     y - - *
+            //
+            // Here we see that the final edit distance will be "y+3".  Again, if the edit 
+            // Distance threshold is less than 3, then could never contribute a path.
+            //
+            // So, if we had an edit distance threshold of 3, then the range around that
+            // bottom diagonal that we should consider checking is:
+            //
+            //           s u n d a y
+            //      ----------------
+            //      |∞ ∞ ∞ ∞ ∞ ∞ ∞ ∞
+            //      |∞ 0 1 2 3 4 5 6
+            //    s |∞ 1 | |
+            //    a |∞ 2 | | |
+            //    t |∞ 3 ` | | |
+            //    u |∞ 4 - ` | | |
+            //    r |∞ 5 - - ` | | |
+            //    d |∞ 6 - - - ` | |
+            //    a |∞ 7   - - - ` |
+            //    y |∞ 8     - - - *
+            //
+            // Now, also consider that it will take a minimum of targetLength-sourceLength edits 
+            // just to move to the lower diagonal from the upper diagonal.  That leaves
+            // 'threshold - (targetLength - sourceLength)' edits remaining.  In this example, that
+            // means '3 - (8 - 6)' = 1.  Because of this our lower diagonal offset is capped at:
+            //
+            //           s u n d a y
+            //      ----------------
+            //      |∞ ∞ ∞ ∞ ∞ ∞ ∞ ∞
+            //      |∞ 0 1 2 3 4 5 6
+            //    s |∞ 1 | |
+            //    a |∞ 2 | | |
+            //    t |∞ 3 ` | | |
+            //    u |∞ 4 - ` | | |
+            //    r |∞ 5   - ` | | |
+            //    d |∞ 6     - ` | |
+            //    a |∞ 7       - ` |
+            //    y |∞ 8         - *
+            //
+            // If we mark the upper diagonal appropriately we see the matrix as:
+            //
+            //           s u n d a y
+            //      ----------------
+            //      |∞ ∞ ∞ ∞ ∞ ∞ ∞ ∞
+            //      |∞ 0 1 2 3 4 5 6
+            //    s |∞ 1 ` |
+            //    a |∞ 2   ` |
+            //    t |∞ 3 `   ` |
+            //    u |∞ 4 - `   ` |
+            //    r |∞ 5   - `   ` |
+            //    d |∞ 6     - `   `
+            //    a |∞ 7       - `  
+            //    y |∞ 8         - *
+            //
+            // Or, effectively, we only need to examine 'threshold - (targetLength - sourceLength)' 
```

Comment:
What if the source length is greater than that of the target? Is the target always the longer one?